File: src/main/java/dev/bernasss12/bebooks/client/gui/ModConstants.java
package dev.bernasss12.bebooks.client.gui;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.Enchantments;
import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;

import java.util.List;
import java.util.Map;

public class ModConstants {
    protected static final int SETTINGS_VERSION = 2;

    public static final List<ItemStack> DEFAULT_CHECKED_ITEMS_LIST = List.of(
        new ItemStack(Items.DIAMOND_SWORD),
        new ItemStack(Items.DIAMOND_PICKAXE),
        new ItemStack(Items.DIAMOND_AXE),
        new ItemStack(Items.DIAMOND_SHOVEL),
        new ItemStack(Items.DIAMOND_HOE),
        new ItemStack(Items.BOW),
        new ItemStack(Items.CROSSBOW),
        new ItemStack(Items.FISHING_ROD),
        new ItemStack(Items.TRIDENT),
        new ItemStack(Items.DIAMOND_HELMET),
        new ItemStack(Items.DIAMOND_CHESTPLATE),
        new ItemStack(Items.DIAMOND_LEGGINGS),
        new ItemStack(Items.DIAMOND_BOOTS),
        new ItemStack(Items.ELYTRA),
        new ItemStack(Items.SHIELD),
        new ItemStack(Items.SHEARS),
        new ItemStack(Items.CARROT_ON_A_STICK),
        new ItemStack(Items.WARPED_FUNGUS_ON_A_STICK),
        new ItemStack(Items.FLINT_AND_STEEL)
    );

    // Default enchantment colors as suggested by twusya on https://www.curseforge.com/minecraft/mc-mods/better-enchanted-books#c47
    public static final Map<Enchantment, Integer> DEFAULT_ENCHANTMENT_COLORS = Map.ofEntries(
        Map.entry(Enchantments.AQUA_AFFINITY, 0x6e7af7),
        Map.entry(Enchantments.BANE_OF_ARTHROPODS, 0x0f5160),
        Map.entry(Enchantments.BLAST_PROTECTION, 0x442e62),
        Map.entry(Enchantments.CHANNELING, 0xaef5ff),
        Map.entry(Enchantments.BINDING_CURSE, 0x274d1e),
        Map.entry(Enchantments.VANISHING_CURSE, 0x171220),
        Map.entry(Enchantments.DEPTH_STRIDER, 0x9cdbff),
        Map.entry(Enchantments.EFFICIENCY, 0xfff164),
        Map.entry(Enchantments.FEATHER_FALLING, 0xfff0d1),
        Map.entry(Enchantments.FIRE_ASPECT, 0xff7516),
        Map.entry(Enchantments.FIRE_PROTECTION, 0xc4aaa5),
        Map.entry(Enchantments.FLAME, 0xff7516),
        Map.entry(Enchantments.FORTUNE, 0xffb65b),
        Map.entry(Enchantments.FROST_WALKER, 0x90b4ff),
        Map.entry(Enchantments.IMPALING, 0xc5133a),
        Map.entry(Enchantments.INFINITY, 0x7b5be7),
        Map.entry(Enchantments.KNOCKBACK, 0x605b60),
        Map.entry(Enchantments.LOOTING, 0xffb65b),
        Map.entry(Enchantments.LOYALTY, 0x6ec4b1),
        Map.entry(Enchantments.LUCK_OF_THE_SEA, 0x4be850),
        Map.entry(Enchantments.LURE, 0xff0000),
        Map.entry(Enchantments.MENDING, 0xcaff61),
        Map.entry(Enchantments.MULTISHOT, 0xffb301),
        Map.entry(Enchantments.PIERCING, 0x337b50),
        Map.entry(Enchantments.POWER, 0xc5133a),
        Map.entry(Enchantments.PROJECTILE_PROTECTION, 0xcccdd5),
        Map.entry(Enchantments.PROTECTION, 0xa5afc4),
        Map.entry(Enchantments.PUNCH, 0x605b60),
        Map.entry(Enchantments.QUICK_CHARGE, 0xff0000),
        Map.entry(Enchantments.RESPIRATION, 0x7ad5ff),
        Map.entry(Enchantments.RIPTIDE, 0xaccff1),
        Map.entry(Enchantments.SHARPNESS, 0xc5133a),
        Map.entry(Enchantments.SILK_TOUCH, 0xffffff),
        Map.entry(Enchantments.SMITE, 0xbf5f2e),
        Map.entry(Enchantments.SOUL_SPEED, 0x41342c),
        Map.entry(Enchantments.SWEEPING, 0xffb301),
        Map.entry(Enchantments.THORNS, 0x560d0b),
        Map.entry(Enchantments.UNBREAKING, 0x5c3350)
    );

    public static final boolean DEFAULT_SHOW_ENCHANTMENT_MAX_LEVEL = false;
    public static final ModConfig.TooltipSetting DEFAULT_TOOLTIP_SETTING = ModConfig.TooltipSetting.ON_SHIFT;
    public static final ModConfig.SortingSetting DEFAULT_SORTING_SETTING = ModConfig.SortingSetting.ALPHABETICALLY;
    public static final boolean DEFAULT_KEEP_CURSES_BELOW = true;
    public static final boolean DEFAULT_COLOR_BOOKS = true;
    public static final boolean DEFAULT_CURSE_COLOR_OVERRIDE = true;
    public static final ModConfig.SortingSetting DEFAULT_COLOR_PRIORITY_SETTING = ModConfig.SortingSetting.ALPHABETICALLY;
    public static final int DEFAULT_BOOK_STRIP_COLOR = 0xc5133a;
    public static final Boolean DEFAULT_GLINT_SETTING = false;
}

File: src/main/kotlin/dev/bernasss12/bebooks/util/ModConstants.kt
package dev.bernasss12.bebooks.util

import dev.bernasss12.bebooks.config.SortingMode
import dev.bernasss12.bebooks.config.TooltipMode
import net.fabricmc.loader.api.FabricLoader
import net.minecraft.enchantment.Enchantment
import net.minecraft.enchantment.Enchantments.*
import net.minecraft.item.ItemStack
import net.minecraft.item.Items.*
import java.nio.file.Path

object ModConstants {
    const val SETTINGS_VERSION: Int = 2
    val CONFIG_DIR: Path = FabricLoader.getInstance().configDir.resolve("bebooks")

    @JvmField // TODO remove when not used by java class
    val DEFAULT_CHECKED_ITEMS_LIST: List<ItemStack> = listOf(
        DIAMOND_SWORD,
        DIAMOND_PICKAXE,
        DIAMOND_AXE,
        DIAMOND_SHOVEL,
        DIAMOND_HOE,
        BOW,
        CROSSBOW,
        FISHING_ROD,
        TRIDENT,
        DIAMOND_HELMET,
        DIAMOND_CHESTPLATE,
        DIAMOND_LEGGINGS,
        DIAMOND_BOOTS,
        ELYTRA,
        SHIELD,
        SHEARS,
        CARROT_ON_A_STICK,
        WARPED_FUNGUS_ON_A_STICK,
        FLINT_AND_STEEL
    ).map(::ItemStack)

    // Default enchantment colors as suggested by twusya on https://www.curseforge.com/minecraft/mc-mods/better-enchanted-books#c47
    @JvmField // TODO remove when not used by java class
    val DEFAULT_ENCHANTMENT_COLORS: Map<Enchantment, Int> = mapOf(
        AQUA_AFFINITY to 0x6e7af7,
        BANE_OF_ARTHROPODS to 0x0f5160,
        BLAST_PROTECTION to 0x442e62,
        CHANNELING to 0xaef5ff,
        BINDING_CURSE to 0x274d1e,
        VANISHING_CURSE to 0x171220,
        DEPTH_STRIDER to 0x9cdbff,
        EFFICIENCY to 0xfff164,
        FEATHER_FALLING to 0xfff0d1,
        FIRE_ASPECT to 0xff7516,
        FIRE_PROTECTION to 0xc4aaa5,
        FLAME to 0xff7516,
        FORTUNE to 0xffb65b,
        FROST_WALKER to 0x90b4ff,
        IMPALING to 0xc5133a,
        INFINITY to 0x7b5be7,
        KNOCKBACK to 0x605b60,
        LOOTING to 0xffb65b,
        LOYALTY to 0x6ec4b1,
        LUCK_OF_THE_SEA to 0x4be850,
        LURE to 0xff0000,
        MENDING to 0xcaff61,
        MULTISHOT to 0xffb301,
        PIERCING to 0x337b50,
        POWER to 0xc5133a,
        PROJECTILE_PROTECTION to 0xcccdd5,
        PROTECTION to 0xa5afc4,
        PUNCH to 0x605b60,
        QUICK_CHARGE to 0xff0000,
        RESPIRATION to 0x7ad5ff,
        RIPTIDE to 0xaccff1,
        SHARPNESS to 0xc5133a,
        SILK_TOUCH to 0xffffff,
        SMITE to 0xbf5f2e,
        SOUL_SPEED to 0x41342c,
        SWEEPING to 0xffb301,
        THORNS to 0x560d0b,
        UNBREAKING to 0x5c3350
    )

    const val DEFAULT_SHOW_ENCHANTMENT_MAX_LEVEL: Boolean = false
    val DEFAULT_TOOLTIP_MODE: TooltipMode = TooltipMode.ON_SHIFT
    val DEFAULT_SORTING_MODE: SortingMode = SortingMode.ALPHABETICALLY
    const val DEFAULT_KEEP_CURSES_BELOW: Boolean = true
    const val DEFAULT_COLOR_BOOKS: Boolean = true
    const val DEFAULT_CURSE_COLOR_OVERRIDE: Boolean = true
    val DEFAULT_COLOR_MODE: SortingMode = SortingMode.ALPHABETICALLY

    @Suppress("MayBeConstant") // TODO Cannot be constant because it is a JvmField
    @JvmField // TODO remove when not used by java class
    val DEFAULT_BOOK_STRIP_COLOR: Int = 0xc5133a
    const val DEFAULT_GLINT_SETTING: Boolean = false
}